### PR TITLE
docs: update popover custom target story

### PR DESCRIPTION
**Pull request by spmonahan**
## Previous Behavior

Previously the example relied on React refs as dependencies to a `useEffect()` call but these will never trigger the effect. The example still works but when trying to extrapolate from it and target elements that appear after the first render, the example breaks down because the effect will not be triggered.

## New Behavior

This example takes advantage of the fact that [useMergedRefs()](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-utilities/src/hooks/useMergedRefs.ts) supports functions as refs so we can trigger the effect when the refs are set.

Using `setState()` was also considered but this will always trigger a re-render, negatively affecting performance so this pattern, while perhaps useful in some cases, is not a good example for best practices documentation. `setState()` is the approach demonstrated in the positioning docs ([link](https://react.fluentui.dev/?path=/docs/concepts-developer-positioning-components--default#anchor-to-target))

The `useEffect()` could have its dependencies removed so it would be called every render which should allow it to function as needed with less of a performance overhead than the `setState()` approach. This may be a better approach for documentation.

## Related Issue(s)

This PR is motivated by user feedback.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Avatar-mount_base.html">625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Avatar-mount.html">653</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Button-mount_base.html">301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Button-mount.html">308</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Field-mount_base.html">1166</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Field-mount.html">1115</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProvider-mount_base.html">706</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProvider-mount.html">693</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProviderWithTheme-mount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">32</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">36</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/MakeStyles-mount_base.html">887</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/MakeStyles-mount.html">875</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Persona-mount_base.html">1791</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/Persona-mount.html">1693</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/SpinButton-mount_base.html">1414</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/SpinButton-mount.html">1394</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/SwatchPicker-mount_base.html">1543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31374/perf-test-react-components/SwatchPicker-mount.html">1519</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx [hunk 1]
```diff
@@ -1,5 +1,13 @@
 import * as React from 'react';
```
Comment (fabricteam) [inline packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31374&commitId=190a5d53fcecfc81c2f28a91e133d24b98a6b339&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|3| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx [hunk 2]
```diff
@@ -24,16 +32,17 @@ const ExampleContent = () => {
...
-  const buttonRef = React.useRef<HTMLButtonElement>(null);
-  const positioningRef = React.useRef<PositioningImperativeRef>(null);
-  const styles = useStyles();
-
-  React.useEffect(() => {
+  const updateTarget = React.useCallback(() => {
     if (buttonRef.current) {
       positioningRef.current?.setTarget(buttonRef.current);
     }
   }, [buttonRef, positioningRef]);
```
Comment (layershifter) [inline packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:39]:
```suggestion
  const positioningRef = React.useRef<PositioningImperativeRef>(null);
  const buttonRef = React.useCallback((el: HTMLButtonElement | null) => {
    positioningRef.current?.setTarget(el)
  }, [positioningRef]);
```

React accepts functions as refs, that should enough 🐱

File: packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx [hunk 3]
```diff
@@ -24,16 +32,17 @@ const ExampleContent = () => {
...
 
+  const buttonRef = useMergedRefs<HTMLButtonElement>(updateTarget);
+  const positioningRef = useMergedRefs<PositioningImperativeRef>(updateTarget);
```
Comment (layershifter) [inline packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:42]:
```suggestion
```

File: packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx (hunk 1 above)
Comment (fabricteam) [inline packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31374&commitId=5f299f9c64373ef23325e95b159b38376c481bcd&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx (hunk 1 above)
Comment (spmonahan) [reply packages/react-components/react-popover/stories/Popover/PopoverAnchorToCustomTarget.stories.tsx:1]: This is noise.

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**